### DISCUSSION TITLE
Remix using SSR  and Load Testing

### DISCUSSION THREAD

**Discussion by VarunTwri** [remix-run/remix#5826] [Q&A]
Remix is not going to use a lot of server resources if everything is SSRed
and if there is a way to load test this or any metrics we have.

**Comment by sergiodxa** [comment]
You can probably use any load testing tool, you will most likely want to perform tests against your own app since most of the time spent during SSR is not on React rendering typically but on your own loaders code getting the data they need.

**Comment by cliffordfajardo** [comment]
Here's is an example Github repo I made doing some HTTP load testing using 2 different HTTP CLI based benchmarking tools several months ago. Github Repo: https://github.com/cliffordfajardo/remix-http-benchmarks



NOTE: @VarunTwri 
1. Your results will vary depending on your machine (CPU, RAM etc)
2. The NodeJS version you use can impact performance. Example: NodeJS v19 enables keep alive HTTP(s) option to true; whereas previous version of node didn't have this on. (What is [Keep Alive](https://connectreport.com/blog/tuning-http-keep-alive-in-node-js/) article)
3. I haven't kept the repo up to date, but thought'd i'd share a reference that might help you or other folks get a glimpse on how you might do it

<a target="_blank" href="https://github.com/cliffordfajardo/remix-http-benchmarks" >
<img height="400" src="https://user-images.githubusercontent.com/6743796/226439902-2e367050-1e7c-410d-a5df-1f54f2558b94.png" />
</a>


## References
- `keep-alive` option for nodejs server: 
    - https://connectreport.com/blog/tuning-http-keep-alive-in-node-js/
    - https://dev.to/gajus/enable-https-keepalive-in-nodejs-2ecn
- HTTP Benchmarking tools used in the repo ^
   - [oha](https://github.com/hatoo/oha)  ----   👈🏽 I prefer this one
   - [autocanon](https://github.com/mcollina/autocannon)
